tests/multi_bluetooth/ble_deepsleep.py

```diff
@@ -0,0 +1,99 @@
+# Test BLE re-connect after peripheral deepsleep.
+# instance0 needs to be a device capable of deepsleep without disconnecting the repl,
+# eg. stm32WB55 Nucleo board with repl on uart port.
+
+from micropython import const
+import time, machine, bluetooth
+
+TIMEOUT_MS = 4000
+
+_IRQ_CENTRAL_CONNECT = const(1)
+_IRQ_CENTRAL_DISCONNECT = const(2)
+_IRQ_PERIPHERAL_CONNECT = const(7)
+_IRQ_PERIPHERAL_DISCONNECT = const(8)
+
+waiting_events = {}
+
+
+def irq(event, data):
+    if event == _IRQ_CENTRAL_CONNECT:
+        print("_IRQ_CENTRAL_CONNECT")
+        waiting_events[event] = data[0]
+    elif event == _IRQ_CENTRAL_DISCONNECT:
+        print("_IRQ_CENTRAL_DISCONNECT")
+    elif event == _IRQ_PERIPHERAL_CONNECT:
+        print("_IRQ_PERIPHERAL_CONNECT")
+        waiting_events[event] = data[0]
+    elif event == _IRQ_PERIPHERAL_DISCONNECT:
+        print("_IRQ_PERIPHERAL_DISCONNECT")
+
+    if event not in waiting_events:
+        waiting_events[event] = None
+
+
+def wait_for_event(event, timeout_ms):
+    t0 = time.ticks_ms()
+    while time.ticks_diff(time.ticks_ms(), t0) < timeout_ms:
+        if event in waiting_events:
+            return waiting_events.pop(event)
+        machine.idle()
+    raise ValueError("Timeout waiting for {}".format(event))
+
+
+# Acting in peripheral role.
+def instance0():
+    global ble, irq
+    multitest.globals(BDADDR=ble.config("mac"))
+    print("gap_advertise")
+    ble.gap_advertise(20_000, b"\x02\x01\x06\x04\tMPY")
+    multitest.next()
+    # Wait for central to connect, then wait for it to disconnect.
+    wait_for_event(_IRQ_CENTRAL_CONNECT, TIMEOUT_MS)
+    wait_for_event(_IRQ_CENTRAL_DISCONNECT, TIMEOUT_MS)
+
+    with multitest.will_reboot(resume="0_resume", delay_ms=100):
+        machine.deepsleep(100)
+
+
+# Acting in peripheral role.
+def instance0_resume():
+    # continue here after reboot
+    print("restarted")
+    global ble, irq
+    multitest.globals(BDADDR=ble.config("mac"))
```

Comment:
this line should be removed:
- it's not needed because it's already set in `instance0()`, and it'll be the same value
- `instance1()` will not pick up the changed value (if it were to change) because it's still running the code with the old value injected into the templace script
- it appears in the expected output when it should not (because the BDADDR changes with a different MCU)